src/state/assets/userAssets.ts

```diff
@@ -270,9 +272,11 @@ export const userAssetsStore = createRainbowStore<UserAssetsState>(
       set(() => {
         const idsByChain = new Map<UserAssetFilter, UniqueId[]>();
         const unsortedChainBalances = new Map<ChainId, number>();
+        const isMap = userAssets instanceof Map;
+        const userAssetsArray = isMap ? Array.from(userAssets.values()) : userAssets;
 
-        userAssets.forEach(asset => {
-          const balance = Number(asset.native.balance.amount) ?? 0;
+        userAssetsArray.forEach(asset => {
+          const balance = Number(asset.native?.balance?.amount ?? asset.balance?.amount) ?? 0;
```

Comment:
I'm not sure if native balance will work just the same as `asset.balance`. This is something to look into.